src/database/operations/assert/assertProposalAuthorization.ts

```diff
@@ -1,14 +1,18 @@
 import { db } from '../../db';
 import { NotFoundError } from '../../../errors';
-import type { CheckResult } from '../../../types';
+import type { AuthContext, CheckResult } from '../../../types';
 
 export const assertProposalAuthorization = async (
 	id: number,
-	userId: number,
+	authContext: AuthContext,
 ): Promise<void> => {
+	const { id: userId } = authContext.user;
+	const { isAdministrator } = authContext.role;
```

Comment:
FWIW, you can do deep destructured assignment more idiomatically like this:
```suggestion
	const { role: { isAdministrator } } = authContext;
```

Reply:
fiiiiiiiiiiiiiiine